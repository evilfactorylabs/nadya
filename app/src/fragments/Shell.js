import React, { Component } from 'react'
import Fab from '@material-ui/core/Fab'
import AddIcon from '@material-ui/icons/Add'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardHeader from '@material-ui/core/CardHeader'
import Avatar from '@material-ui/core/Avatar'
import SimpleIcons from 'simple-icons-react-component'

import AuthHOC from 'components/Auth'

import DialogLazy from '../components/Lazy'
import {
  listSubscription,
  listenUpdate,
  unlistenUpdate
} from 'services/subscription'

import { navigate } from '@reach/router'
import { toCurrency } from 'utils'

const classes = {
  card: {
    minWidth: 275
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)'
  },
  title: {
    fontSize: 14
  },
  pos: {
    marginBottom: 12
  },
  fab: {
    position: 'fixed',
    bottom: 20,
    right: 20
  },
  content: {
    padding: '2rem'
  },
  container: {
    maxHeight: '90vh',
    overflowY: 'auto'
  }
}

export default class App extends Component {
  state = {
    isDialogOpen: false,
    subscriptions: []
  }

  componentDidMount() {
    listSubscription().then(doc => {
      this.setState({ subscriptions: doc.rows })
    })

    listenUpdate(doc => {
      this.setState({
        subscriptions: [doc, ...this.state.subscriptions]
      })
    })
  }

  _handleDialogOpen = () => {
    this.setState({ isDialogOpen: true })
  }

  _handleDialogClose = () => {
    this.setState({ isDialogOpen: false })
  }

  componentWillUnmount() {
    unlistenUpdate()
  }

  render() {
    return AuthHOC(() => (
      <div className='App'>
        {!this.state.subscriptions.length && (
          <header style={classes.content}>
            <p>I'm empty state</p>
          </header>
        )}
        <div style={classes.container}>
          {this.state.subscriptions
            .filter(({ doc }) => Object.keys(doc).length !== 0)
            .map(({ doc }) => (
              <Card className={classes.card} key={doc._id}>
                <CardActionArea onClick={() => navigate(`/${doc._id}/`)}>
                  <CardHeader
                    avatar={
                      <Avatar
                        style={{ backgroundColor: 'transparent' }}
                        aria-label='Recipe'
                        className={classes.avatar}
                      >
                        <SimpleIcons name={doc.title} />
                      </Avatar>
                    }
                    title={doc.title}
                    subheader={toCurrency(doc.cost)}
                  />
                </CardActionArea>
              </Card>
            ))}
        </div>
        <Fab
          onClick={this._handleDialogOpen}
          color='primary'
          aria-label='Add'
          style={classes.fab}
        >
          <AddIcon />
        </Fab>
        <DialogLazy
          component='./Dialog'
          open={this.state.isDialogOpen}
          handleClose={this._handleDialogClose}
        />
      </div>
    ))
  }
}
