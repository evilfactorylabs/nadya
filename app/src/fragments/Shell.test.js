import Shell from './Shell'

afterAll(async () => {
  await cleanup()
})

describe('Fragments: Shell.js', () => {
  test('Fab click & show fullscreen dialog', () => {
    const { container } = render(<Shell />)

    expect(container).toBeInTheDocument()
  })
})
