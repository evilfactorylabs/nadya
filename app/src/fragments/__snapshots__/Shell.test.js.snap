// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments: Shell.js should render without crash 1`] = `
<div>
  <div
    class="App"
  >
    <header
      style="padding: 2rem;"
    >
      <p>
        I'm empty state
      </p>
    </header>
    <div
      style="max-height: 90vh; overflow-y: auto;"
    />
    <button
      class="MuiButtonBase-root MuiFab-root MuiFab-primary"
      style="position: fixed; bottom: 20px; right: 20px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
